src/utils/fillwidth.tsx

```diff
@@ -0,0 +1,77 @@
+import { reduce } from "lodash"
+
+const fillwidthDimensions = (items, containerWidth, gutter, targetHeight) => {
+  /**
+   * Scales an image object proportionally based on a direction (either -1 or 1)
+   * @param img a dimension object that references an artwork image
+   * @param dir the direction we need to scale an image, either -1 or 1
+   */
+  const resizeHeight = (img, dir) => {
+    img.width += img.width / img.height * dir
+    img.height += dir
```

Comment:
Looking over this again (this was copied over from `artwork/fillwidth.tsx`) -- I _think_ it might be ok to mutate here. The fillwidth starts by creating a `dimensions` array with a best guess of width/height and the point of `resizeHeight` is for the images in `dimensions` to keep adjusting themselves until an acceptable difference is found in a while loop. If we don't mutate, we'd have to also reconstruct the `dimensions` array for each adjustment.  

```
  // Keep looping until we get an acceptable width difference
  while (true) {
    for (let img of dimensions) {
      resizeHeight(img, dir)
      if (widthDiff(dimensions) > 1) {
        break
      }
    }
    if (widthDiff(dimensions) > 1) {
      break
    }
  }
```